tests/contrib/django/test_middleware.py

```diff
@@ -63,6 +63,21 @@ def test_middleware_trace_function_based_view(self):
         eq_(span.get_tag('http.url'), '/fn-view/')
         eq_(span.resource, 'tests.contrib.django.app.views.function_view')
 
+    def test_middleware_trace_error_500(self):
+        # ensures that all error 500 are well traced
+        url = reverse('error-500')
+        response = self.client.get(url)
+        eq_(response.status_code, 500)
+
+        # check for spans
+        spans = self.tracer.writer.pop()
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.get_tag('http.status_code'), '500')
+        eq_(span.get_tag('http.url'), '/error-500/')
+        eq_(span.resource, 'tests.contrib.django.app.views.error_500')
+        assert "ZeroDivisionError: integer division or modulo by zero" in span.get_tag('error.stack')
```

Comment:
You can stick with this implementation and we may think about the one I wrote above in a second step (maybe later). If we need to get more details about middlewares, including exceptions, we need to wrap all registered middlewares. This will provide better instrumentation, especially because knowing the time spent in middlewares is really useful for a Django developer.